oxd-gen-client/src/test/java/io/swagger/client/api/RegisterSiteTest.java

```diff
@@ -0,0 +1,112 @@
+package io.swagger.client.api;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import io.swagger.client.ApiResponse;
+import io.swagger.client.model.RegisterSiteParams;
+import io.swagger.client.model.RegisterSiteResponse;
+import io.swagger.client.model.RegisterSiteResponseData;
+import io.swagger.client.model.UpdateSiteParams;
+import io.swagger.client.model.UpdateSiteResponse;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import org.xdi.oxauth.model.common.GrantType;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+
+import static io.swagger.client.api.Tester.api;
+import static io.swagger.client.api.Tester.getAuthorization;
+import static io.swagger.client.api.Tester.notEmpty;
+import static junit.framework.Assert.*;
+
+/**
+ * @author Yuriy Zabrovarnyy
+ * @author Shoeb Khan
+ * @version 07/26/2018
+ */
+
+@Test
+public class RegisterSiteTest {
+
+    private String oxdId = null;
+
+    @Parameters({"host", "opHost", "redirectUrl", "logoutUrl", "postLogoutRedirectUrl"})
+    @Test
+    public void register(String host, String opHost, String redirectUrl, String postLogoutRedirectUrl, String logoutUrl) throws Exception {
+        try {
+            DevelopersApi client = api();
+            //
+            RegisterSiteResponseData resp = registerSite(client, opHost, redirectUrl, postLogoutRedirectUrl, logoutUrl);
+            assertNotNull(resp);
+            notEmpty(resp.getOxdId());
+            //
+            // more specific site registration
+            final RegisterSiteParams params = new RegisterSiteParams();
+            //commandParams.setProtectionAccessToken(setupClient.getClientRegistrationAccessToken());
+            params.setOpHost(opHost);
+            params.setAuthorizationRedirectUri(redirectUrl);
+            params.setPostLogoutRedirectUri(postLogoutRedirectUrl);
+            params.setClientFrontchannelLogoutUris(Lists.newArrayList(logoutUrl));
+            params.setRedirectUris(Lists.newArrayList(redirectUrl));
+            params.setAcrValues(new ArrayList<String>());
+            params.setScope(Lists.newArrayList("openid", "profile"));
+            params.setGrantTypes(Lists.newArrayList("authorization_code"));
+            params.setResponseTypes(Lists.newArrayList("code"));
+
+            resp = client.registerSite(params).getData();
+            assertNotNull(resp);
+            assertNotNull(resp.getOxdId());
+            oxdId = resp.getOxdId();
+        } catch (Exception ex) {
+            fail(ex.getMessage());
```

Comment:
why do we need try/catch block here? let go exception go outside with full stack trace.